fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -1548,7 +1548,7 @@ private boolean updateCatalogAfterVisible(TransactionState transactionState, Dat
 
     // the write lock of database has been hold
     private boolean updateCatalogAfterVisibleBatch(TransactionStateBatch transactionStateBatch, Database db) {
-        Table table = db.getTable(transactionStateBatch.getTableId());
+        Table table = db.getTable(transactionStateBatch.getTableId().get(0));
```

Comment:
`TransactionStateBatch.getTableId()` may return an empty list, according to your implementation, should check the list size before calling `get(0)`

Reply:
The transaction here is probably a multi-table transaction, right? If so, it's possible that some of the tables have already been dropped, but here it's just determining whether the first table exists, potentially leading to missing processing those that still exist, right?

Reply:
The transaction with multi-table will not go through the logic of batch，it will degenerate into the original single publish mode，in another word，txns in transactionStateBatch in`updateCatalogAfterVisibleBatch` must not involve multi-table.

Reply:
If so, maybe we can change the return type of `TransactionStateBatch.getTableId()` from `List<Long>` to `Long`

Reply:
If so, can we change the return type of `TransactionStateBatch.getTableId()` from `List<Long>` to `Long`?